src/FormMigration/Steps/FormFieldManager.php

```diff
@@ -423,40 +429,39 @@ private function insertBlock($block): void
     }
 
     /**
+     * @unreleased Fixed missing conditionalLogic attribute on custom fields.
      * @since 3.0.0-rc.7
      */
-
     private function mapConditionalLogicToBlocks(): void
     {
         foreach ($this->fieldBlockRelationships as $item) {
             ['field' => $field, 'block' => $block] = $item;
 
-            if (!array_key_exists('control_field_visibility', $field) || $field['control_field_visibility'] !== 'on') {
-                if ($block->name !== 'givewp/section') {
-                    $block->setAttribute('conditionalLogic', [
-                        'enabled' => false,
-                        'action' => 'show',
-                        'boolean' => 'and',
-                        'rules' => [],
-                    ]);
-                }
+            // Initialize conditional logic support for custom fields.
+            $block->setAttribute('conditionalLogic', [
```

Comment:
Shouldn't we be checking if conditional logic is enabled first before adding the attribute? I find it strange that we're adding conditional logic that does nothing.

Reply:
The attribute is an object and doubles as a flag for the block inspector controls. Otherwise you couldn't enable conditional logic for the migrated field.

Reply:
I updated the comment to clarify _why_, see https://github.com/impress-org/givewp/pull/7030/commits/b88d65562ff50ac6e75fe97578d7a7c7cc4ad274.